### PR TITLE
2.0 Use `null` instead of `false` as return type in factories

### PR DESCRIPTION
**Pull request by gchtr**
**Ticket**: #2561

### CODE REVIEW

File: docs/v2/upgrade-guides/2.0.md
Comment (jarednova) [inline docs/v2/upgrade-guides/2.0.md:207]:
```suggestion
`Timber::get_post()` returned `false` if no post could be found in 1.x. Now, that function will return `null` if no post could be found. We recommend using `null` instead of `false` for when no objects are found to match modern PHP practices.
```

**Review by jarednova — CHANGES_REQUESTED** [review]
My only suggestion is to add a more universal note about `null` > `false` in the Upgrade Guide as we're also showing this as the prefered style like in `docs/v2/guides/extending-timber.md:287`

**Comment by gchtr** [comment]
Thanks @jarednova. I committed your suggestion. With a final approval, this is ready to merge.

**Review by jarednova — APPROVED** [review]

**Comment by jarednova** [comment]
Final approval granted, please merge away!

**Review by nlemoine — APPROVED** [review]

**Comment by gchtr** [comment]
I’m waiting for the changes in #2517 to be merged before merging this PR in.

**Comment by jarednova** [comment]
Thanks @gchtr. To confirm where we are on #2517, is the last pending edit the one described in this [comment/discussion](https://github.com/timber/timber/pull/2517#pullrequestreview-900994888), or are there any other items requiring resolution?

**Pull request merged**
